ansible/roles/test/tasks/tacacs_test.yml

```diff
@@ -0,0 +1,83 @@
+- name: Set variables for TACACS+ test
+  set_fact:
+    loopback_ip: "100.1.1.1"
+
+# Test TACACS+ login authentication
+- name: Enable TACACS+ Authentication
+  become: true
+  shell: config aaa authentication login tacacs+ local
+
+- name: Check if pam configuration ok
+  become: true
+  shell: "grep 'pam_tacplus.so server={{ tacacs_server }}' /etc/pam.d/common-auth-sonic"
+  register: tacplus_pam_module
+  failed_when: '"secret={{ tacacs_passkey }}" not in tacplus_pam_module.stdout'
+
+- name: Check if TACACS+ user login ok
+  shell: "sshpass -p {{ tacacs_passwd }} ssh {{ tacacs_username }}@{{ ansible_host }} whoami"
+  connection: local
+  become: no
+  register: login_result
+  failed_when: login_result.stdout != "{{ tacacs_username }}"
+
+# Test failthrough mechanism
+- name: Config local authentication prior to TACACS+ authentication
+  become: true
+  shell: config aaa authentication login local tacacs+
+
+- name: Disable fail-through mechanism
+  become: true
+  shell: config aaa authentication failthrough disable
+
+- name: Check if TACACS+ user login fail
+  shell: "sshpass -p {{ tacacs_passwd }} ssh {{ tacacs_username }}@{{ ansible_host }} whoami"
+  connection: local
+  become: no
+  register: login_result
+  failed_when: login_result.stdout == "{{ tacacs_username }}"
+
+- name: Restore fail-through mechanism
+  become: true
+  shell: config aaa authentication failthrough default
+
+# Test source address
```

Comment:
```
admin@str-s6000-on-4:~$ sudo config tacacs src_ip
Usage: config tacacs [OPTIONS] COMMAND [ARGS]...

Error: No such command "src_ip".
```

Reply:
since src_ip is not supported in current version, I suggest to remove the src_ip test.

Reply:
Already removed it.